src/OrchardCore.Modules/OrchardCore.Users/ViewModels/RegisterExternalLoginViewModel.cs

```diff
@@ -1,13 +1,21 @@
+using System;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Localization;
-using MimeKit;
+using OrchardCore.Email;
 
 namespace OrchardCore.Users.ViewModels
 {
     public class RegisterExternalLoginViewModel : IValidatableObject
     {
+        private readonly IEmailAddressValidator _emailAddressValidator;
+
+        public RegisterExternalLoginViewModel(IEmailAddressValidator emailAddressValidator)
+        {
+            _emailAddressValidator = emailAddressValidator ?? throw new ArgumentNullException(nameof(emailAddressValidator));
```

Comment:
@hishamco you can't pass the `IEmailAddressValidator` into a constructor of a view model.

You need to resolve it from the `ValidationContext`

Reply:
It's passed from where it called, no problem I will address your feedback now, I think I did it two times